src/utils/daemon.js

```diff
@@ -1,5 +1,62 @@
 import IPFSFactory from 'ipfsd-ctl'
 import logger from './logger'
+import { fsckError } from './errors'
+import { join } from 'path'
+import fs from 'fs-extra'
+import { spawnSync } from 'child_process'
+import findExecutable from 'ipfsd-ctl/src/utils/find-ipfs-executable'
+
+function repoFsck (path) {
+  const exec = findExecutable('go', join(__dirname, '..'))
+  spawnSync(exec, ['repo', 'fsck'], {
+    env: {
+      ...process.env,
+      IPFS_PATH: path
+    }
+  })
+}
+
+async function cleanup (path) {
+  const apiFile = join(path, 'api')
+
+  if (!await fs.pathExists(apiFile)) {
+    return true
+  }
+
+  const cfgFile = join(path, 'config')
+  const cfg = await fs.readJSON(cfgFile)
+  const cfgAddr = cfg.Addresses ? cfg.Addresses.API : ''
+  const addr = (await fs.readFile(apiFile)).toString()
```

Comment:
File often ends with newline character because people create it manually with `echo "/ip4/127.0.0.1/tcp/5001" > ~/.ipfs/api`  :)

```suggestion
  const addr = (await fs.readFile(apiFile)).toString().trim()
```